apps/src/templates/progress/GoogleClassroomShareButton.jsx

```diff
@@ -0,0 +1,103 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import Button from '../Button';
+import i18n from '@cdo/locale';
+
+// https://developers.google.com/classroom/brand
+const styles = {
+  label: {
+    paddingLeft: 16,
+    fontFamily: 'Roboto',
+    fontWeight: 700
+  },
+  container: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'start'
+  }
+};
+class GoogleClassroomShareButton extends React.Component {
+  static propTypes = {
+    buttonId: PropTypes.string.isRequired,
+    url: PropTypes.string.isRequired,
+    itemtype: PropTypes.string.isRequired,
+    title: PropTypes.string,
+    height: PropTypes.number,
+    courseid: PropTypes.number
+  };
+
+  constructor(props) {
+    super(props);
+    this.loadApi = this.loadApi.bind(this);
+    this.waitForGapi = this.waitForGapi.bind(this);
+    this.renderButton = this.renderButton.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.gapiReady()) {
+      this.renderButton();
+    } else {
+      this.loadApi();
+    }
+  }
+
+  /* The Google Classroom Share Button is only available through their js api,
+   * so we have to add it to the page on load. Each instance of this
+   * component will wait until the api is ready before rendering the button,
+   * but only the first will kick of the loading process.
+   */
+  loadApi() {
+    if (!document.getElementById('gapi')) {
+      window.___gcfg = {
+        parsetags: 'explicit'
+      };
+
+      const gapi = document.createElement('script');
+      gapi.src = 'https://apis.google.com/js/platform.js';
+      gapi.id = 'gapi';
+      gapi.addEventListener('load', this.waitForGapi);
+      document.body.appendChild(gapi);
+    } else {
+      this.waitForGapi();
+    }
+  }
+
+  waitForGapi() {
+    if (this.gapiReady()) {
+      this.renderButton();
+    } else {
+      setTimeout(() => {
+        this.waitForGapi();
+      }, 100);
+    }
+  }
+
+  gapiReady = () =>
```

Comment:
NIT: Looks like you don't need an arrow function here. Should just be able to do gapiReady() { ... }

Reply:
thanks for calling this out. i thought arrow functions were preferred for single-statement functions. [this article](https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1) set me straight on that!

i also discovered that arrow functions will auto-bind `this`, and thought that was nifty and started using it extensively in my follow-up metrics PR. but when i double-checked our style guide to see what is the preferred method of binding, i discovered that using arrow functions is considered ["very bad"](https://github.com/airbnb/javascript/tree/master/react#methods)!